### DISCUSSION TITLE
Updating Microsoft.WindowsAppSDK for a WinUI3 desktop app (unpackaged)

### DISCUSSION THREAD

**Discussion by rcohn** [microsoft/WindowsAppSDK#2449] [Q&A] [answered]
Hello -

I tried to update my WinUI3 desktop unpackaged app from Microsoft.WindowsAppSDK,1.0, the version I have been using, to 1.03 - the latest stable release.  The app crashes right off when rebuilding with 1.03.  

Not much output:
`'Categories.exe' (CoreCLR: DefaultDomain): Loaded 'C:\Samples\WinUI3\Categories2\bin\x64\Debug\net5.0-windows10.0.19041.0\win10-x64\System.Private.CoreLib.dll'. 
'Categories.exe' (CoreCLR: clrhost): Loaded 'C:\Samples\WinUI3\Categories2\bin\x64\Debug\net5.0-windows10.0.19041.0\win10-x64\Categories.dll'. Symbols loaded.
'Categories.exe' (CoreCLR: clrhost): Loaded 'C:\Samples\WinUI3\Categories2\bin\x64\Debug\net5.0-windows10.0.19041.0\win10-x64\System.Runtime.dll'. 
'Categories.exe' (CoreCLR: clrhost): Loaded 'C:\Samples\WinUI3\Categories2\bin\x64\Debug\net5.0-windows10.0.19041.0\win10-x64\Microsoft.WindowsAppRuntime.Bootstrap.Net.dll'. 
Exception thrown: 'System.Runtime.InteropServices.COMException' in Microsoft.WindowsAppRuntime.Bootstrap.Net.dll
The program '[16396] Categories.exe' has exited with code 2148734499 (0x80131623).`

This appears to be possibly a mismatch in the WindowsAppSDK runtime distribution and the version of WindowsAppSDK being used in the app.  I verified this crash behavior with two different WinUI3 apps.

So my question is what must be done when updating my WinUI3 app to a more recent version of WindowsAppSDK?

Thank you.

**Comment by castorix** [comment, marked as answer]
If you use the Nuget Package 1.03, did you install the 1.03 Runtime ?

**Reply by rcohn** [reply]
Hi @castorix -

I assumed that I would need the 1.03 runtime, but I could not seem to find it.  So I tried using the 1.03 with the 1.0 runtime that I had been using.  Could you direct me to where I can get the 1.03 runtime?

Thanks very much.

**Reply by rcohn** [reply]
OK.  I found the 1.03 redistributable at https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/downloads

When I install the Microsoft.WindowsAppSDK 1.03 from NuGet, and use the 1.03 redistributable installer obtained from the above link, then I am not seeing the crash now.

Thanks for your assistance.
